### PR TITLE
feat: added /problems routes

### PR DESCRIPTION
**Pull request by spierceVR**
- [x] Problem model
- [x] Problem controller
  - [x] GET ``/problems`` route (finds user by ID)
  - [x] GET ``/problems/:id`` route (find multiple users with count and offset params)
- [x] Problem service
- [x] Tested [using postman ](https://go.postman.co/workspace/New-Team-Workspace~e3a5f4b4-0f83-4394-a88d-ea1e55d2f504/collection/21181278-4e012e1f-870f-4975-aedc-a220b30037a4?action=share&creator=21181278)

### CODE REVIEW

**Review by Anphs — APPROVED** [review]
Awesome!

File: central-service/controllers/problem.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+type ProblemController struct {
+	problemService *services.ProblemService
+}
+
+func InitializeProblemController(service *services.ProblemService) ProblemController {
+	return ProblemController{service}
+}
+
+// An endpoint containing logic used to create a new problem based on provided parameters
+func (controller *ProblemController) CreateNewProblemEndpoint(c echo.Context) error {
```
Comment (DedsecKnight) [inline central-service/controllers/problem.go]: What is the use case for this endpoint? Are we using this endpoint to manually add a leetcode problem into our database?

File: central-service/services/problem.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	return &problem, nil
+}
+
+// Function to update problem with specified id
+func (service *ProblemService) UpdateProblemData(problemId uint, problemData *models.Problem) (*models.Problem, error) {
+	searchResult, err := service.FindProblemByProblemID(problemId)
+	if searchResult == nil {
+		return nil, gorm.ErrRecordNotFound
+	}
+	if err != nil {
```
Comment (DedsecKnight) [inline central-service/services/problem.go:45]: This should be checked before checking whether `searchResult == nil`

**Review by DedsecKnight — CHANGES_REQUESTED** [review]
just need to address a few small things, but everything else lgtm!

File: central-service/controllers/problem.go (hunk 1 above)
Comment (spierceVR) [reply central-service/controllers/problem.go]: Good point, we don't actually need this to add problems to our DB. 
We also don't need the update problems endpoint for the same reason, so I'll remove that as well.

And I remembered we also want a fetch all problems endpoint so I'll work on that also.

TODO for me:
- [x] Remove Create endpoint
- [x] Remove Update endpoint
- [x] Add fetch all endpoint (with pagination)

**Review by DedsecKnight — APPROVED** [review]
lgtm :+1:

**Pull request merged**